compiler/src/dmd/expressionsem.d

```diff
@@ -9251,13 +9251,14 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
 
         // Check for unsafe casts
-        if (!isSafeCast(ex, t1b, tob))
-        {
-            if (sc.setUnsafe(false, exp.loc, "cast from `%s` to `%s` not allowed in safe code", exp.e1.type, exp.to))
-            {
-                return setError();
-            }
-        }
+        auto r = isSafeCast(ex, t1b, tob);
+        if (r == -1 && sc.setUnsafePreview(FeatureState.default_, false, exp.loc,
+                "cast from `%s` to `%s` not allowed in safe code", exp.e1.type, exp.to))
```

Comment:
If I understand correctly, it is downcasts are unsafe? If so, can you make the error message convey that?